frontend/catalyst/api_extensions/control_flow.py

```diff
@@ -742,6 +761,78 @@ def __call__(self):
             return self._call_during_interpretation()
 
 
+class CondCallableSingleGateHandler(CondCallable):
+    """
+    Special CondCallable when the conditional body function is a single pennylane gate.
+
+    A usual pennylane conditional call for a gate looks like
+    `qml.cond(x == 42, qml.RX)(theta, wires=0)`
+
+    Since gates are guaranteed to take in arguments (at the very least the wire argument),
+    the usual CondCallable class, which expects the conditional body function to have no arguments,
+    cannot be used.
+    This class inherits from base CondCallable, but wraps the gate in a function with no arguments,
+    and sends that function to CondCallable.
```

Comment:
Could we not just use this strategy all the time, instead of only when the target is a gate class ?